### PR TITLE
[no ticket][risk=no] Fix e2e notebook-ui test

### PR DESCRIPTION
**Pull request by aweng98**
This test can fail when the same notebook is deleted by another test running concurrently or Jenkins delete-workspace script.

### CODE REVIEW

File: e2e/tests/ui/notebook-ui.spec.ts [hunk 1]
```diff
@@ -57,12 +61,18 @@ describe('Notebook and Runtime UI Test', () => {
...
-      logger.info(`Cannot find workspace "${workspaceName}". Test end early.`);
       return false;
     }
 
     // Open workspace.
     await workspaceCard.clickWorkspaceName();
     return true;
   }
+
+  async function openNotebook(page: Page): Promise<DataResourceCard | null> {
```
Comment (jmthibault79) [inline e2e/tests/ui/notebook-ui.spec.ts:72]: how about `openAnyNotebook()` ?

File: e2e/tests/ui/notebook-ui.spec.ts [hunk 2]
```diff
@@ -19,17 +20,20 @@ describe('Notebook and Runtime UI Test', () => {
   test('Notebook name is unique', async () => {
     const existWorkspace = await openWorkspace(page, workspaceName);
     if (!existWorkspace) {
+      logger.info(`Cannot find workspace "${workspaceName}". Test end early.`);
       return;
```
Comment (jmthibault79) [inline e2e/tests/ui/notebook-ui.spec.ts:24]: What's the correct behavior when a test precondition is not met: pass or fail?

File: e2e/tests/ui/notebook-ui.spec.ts [hunk 3]
```diff
@@ -57,12 +61,18 @@ describe('Notebook and Runtime UI Test', () => {
     const workspaceCard = await WorkspaceCard.findCard(page, workspaceName);
```
Comment (jmthibault79) [inline e2e/tests/ui/notebook-ui.spec.ts:61]: I realize that this PR didn't add this, but please remove the comment above this line. It isn't true.

**Review by jmthibault79 — APPROVED** [review]

File: e2e/tests/ui/notebook-ui.spec.ts (hunk 2 above)
Comment (aweng98) [reply e2e/tests/ui/notebook-ui.spec.ts:24]: It's pass. Targeted workspace and notebook will be available after `notebook-create-python.spec` ran.

File: e2e/tests/ui/notebook-ui.spec.ts (hunk 1 above)
Comment (aweng98) [reply e2e/tests/ui/notebook-ui.spec.ts]: Sounds good.

File: e2e/tests/ui/notebook-ui.spec.ts (hunk 3 above)
Comment (aweng98) [reply e2e/tests/ui/notebook-ui.spec.ts:60]: Done.

**Pull request merged**